mne/source_space.py

```diff
@@ -1550,19 +1550,32 @@ def _make_voxel_ras_trans(move, ras, voxel_size):
     return t
 
 
-def _make_discrete_source_space(pos):
+def _make_discrete_source_space(pos, coord_frame='mri'):
     """Use a discrete set of source locs/oris to make src space
 
     Parameters
     ----------
     pos : dict
         Must have entries "rr" and "nn". Data should be in meters.
+    coord_frame : str
+        The coordinate frame in which the positions are given; default: 'mri'.
+        The frame must be one defined in transforms.py:_str_to_frame
 
     Returns
     -------
     src : dict
         The source space.
     """
+    # Check that coordinate frame is a valid string type
+    # (copied from transforms.py:transform_surface_to)
+    if isinstance(coord_frame, string_types):
+        if coord_frame not in _str_to_frame:
+            raise KeyError('coord_frame must be one of %s, not "%s"'
+                           % (list(_str_to_frame.keys()), coord_frame))
+        coord_frame = _str_to_frame[coord_frame]  # now an int
+    else:
+        raise ValueError('coord_frame must be a string')
```

Comment:
You actually probably don't need this type check -- if it's something other than string, the `if coord_frame not in _str_to_frame` will already fail.